### PR TITLE
feat(tokens): Phase A — motion, shadow, surface, fluid-type tokens + tokens-marketing + design-lint

### PR DESCRIPTION
**Pull request by apaksh-gupta**
## Summary

Phase A token expansion across the Amplify Design System.

### tokens-foundation v2.0.1 → v2.1.0
- New JSON tokens: motion (+lazy 700ms duration, linear/bounce easings, stagger 1/2/3), shadows (elevation scale 0–5 + accent sm/md/lg)
- New hand-authored CSS layer at `src/` with canonical `--amplify-*` prefix:
  - `motion.css` — durations, easings, stagger delays, reduced-motion override
  - `shadows.css` — elevation 0–5, coloured accent shadows, dark-mode overrides
  - `surfaces.css` — primary/secondary/tertiary/elevated/overlay aliases
  - `borders.css` — subtle/default/strong/accent + status + widths
  - `fluid-type.css` — `clamp(min, fluid, max)` scale + leading + tracking
- New exports: `./phase-a`, `./motion`, `./shadows`, `./surfaces`, `./borders`, `./fluid-type`

### tokens-marketing v1.0.0 (new package)
Expressive sibling for landing / Step 0 / pricing surfaces. Bigger type scale (hero 56→108px), generous spacing, motion-forward (300/500/800/1200ms), accent-rich gradients. Vars prefixed `--amplify-marketing-*`. Peer-deps tokens-foundation ^2.1.0.

### design-lint v1.0.0 (new package)
ESLint v9 flat-config preset with 3 rules: no-hardcoded-colors, no-raw-spacing, prefer-token-import. Recommended (warn) + strict (error) presets. Node test runner covers all rules.

### Storybook examples
14 new co-located story files in `packages/ui/src/components/{name}/{name}.stories.tsx` covering Avatar, Breadcrumb, Checkbox, Chip, IconButton, Input, MetricCard, Pagination, ProgressBar, Select, Separator, StatusTag, Switch, Toast, Tooltip — 100+ story variants targeting empty/loading/error/disabled/long-content/overflow edge cases.

### CI
Builds tokens-marketing, runs design-lint tests, verifies dist outputs.

## Test plan

- [x] `npm run build` — all 6 workspace packages build green
- [x] `npm run validate` — token consistency passes (0/0)
- [x] `npm run test` — 20 UI tests pass
- [x] design-lint rules: 3/3 pass
- [x] Storybook builds in 5.4s
- [ ] Chromatic visual regression on PR
- [ ] CI pipeline green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/ui/src/components/MetricCard/MetricCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  args: { label: 'Total reach', value: '14,592,318' },
+};
+
+export const ZeroValue: Story = {
+  args: { label: 'Pending approvals', value: 0, subtitle: 'All clear' },
+};
+
+export const Grid: Story = {
+  render: () => (
+    <div style={{ display: 'grid', gridTemplateColumns: 'repeat(3, 1fr)', gap: 16 }}>
```
Comment (zenith-oi) [inline packages/ui/src/components/MetricCard/MetricCard.stories.tsx:75]: **FLAG** — [no-raw-spacing (self-referential)] The Grid story uses `gap: 16` — a raw numeric spacing value — in a JSX style prop: `style={{ display: 'grid', gridTemplateColumns: 'repeat(3, 1fr)', gap: 16 }}`. This is exactly the pattern the new `no-raw-spacing` lint rule flags as invalid (numeric non-zero spacing prop). Other stories have the same pattern (e.g. IconButton.stories.tsx L32 `gap: 8`, Checkbox.stories.tsx L32 `gap: 6`, etc.). Story files shipped in the same PR as the enforcement tooling should demonstrate correct usage — replace with `var(--amp-spacing-*)` tokens or at minimum comment that stories are excluded from the rule. If stories are intentionally excluded from the rule, add a `files` override in index.js to exclude `**/*.stories.*`.

File: packages/ui/src/components/Separator/Separator.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      <span>Duplicate</span>
+      <Separator orientation="vertical" />
+      <span>Delete</span>
+    </div>
+  ),
+};
+
+export const InsideCard: Story = {
+  render: () => (
+    <div style={{ border: '1px solid #e5e5e5', borderRadius: 8, padding: 16, width: 280 }}>
```
Comment (zenith-oi) [inline packages/ui/src/components/Separator/Separator.stories.tsx:40]: **FLAG** — [no-hardcoded-colors (self-referential)] The InsideCard story contains hardcoded hex colors in JSX style props: `border: '1px solid #e5e5e5'`, `color: '#666'`, `color: '#888'`. These are exactly what the new `no-hardcoded-colors` rule flags. Like the spacing issue, stories in the same PR as the enforcement tooling should model correct token usage, not violate the rules being introduced. Replace with `--amplify-border-*` / `--amplify-surface-*` token variables, or explicitly scope stories out of the rule's `files` glob.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**This PR adds three new packages (design-lint, tokens-marketing), expands tokens-foundation with semantic CSS token files and a fluid type scale, and adds Storybook stories for 15 UI components. The work is clean and well-structured overall, with one real issue worth flagging.**

**2 FLAGs** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The prefers-reduced-motion override in both tokens-foundation/src/motion.css and tokens-marketing/src/motion.css is thorough — all duration and stagger tokens are collapsed to 1ms/0ms, and the marketing parallax multipliers are correctly zeroed out. This is easy to forget and gets it right.
- The shadow token refactor in shadows.json is backwards-compatible: the new numeric elevation scale (0–5) is additive, and the old sm/md/lg/sm-dark/md-dark/lg-dark keys are preserved with updated descriptions noting their legacy status. Zero breaking changes for existing consumers.
- The design-lint ESLint plugin correctly implements ESLint v9 flat-config shape from the start, with both a warn-level recommended preset and an error-level strict preset — good forward compatibility.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Comment by apaksh-gupta** [comment]
**Consolidation note (per user direction 2026-04-29):**

`packages/eslint-config` already exists in this repo per CLAUDE.md (`no-hardcoded-colors`, `no-raw-spacing`, `prefer-token-import` rules). The new `packages/design-lint` added in this PR duplicates that scope.

Recommended: before merge, consolidate `packages/design-lint` rules into `packages/eslint-config`, then delete `packages/design-lint`. Single source of truth.

I'll dispatch a follow-up agent to do this consolidation if you'd prefer to merge as-is and clean up after.

File: packages/ui/src/components/MetricCard/MetricCard.stories.tsx (hunk 1 above)
Comment (zenith-oi) [inline packages/ui/src/components/MetricCard/MetricCard.stories.tsx:75]: **FLAG** — [no-raw-spacing (self-referential)] The Grid story uses `gap: 16` (raw numeric pixel value) in a JSX style prop. This is exactly the pattern the newly-shipped `no-raw-spacing` rule flags as a violation. Story files will be linted by the new plugin once `design-lint` is wired into the product ESLint config. Fix: `gap: 'var(--amp-spacing-md)'` or equivalent token. Carried over from previous review — still unresolved.

File: packages/ui/src/components/Separator/Separator.stories.tsx (hunk 2 above)
Comment (zenith-oi) [inline packages/ui/src/components/Separator/Separator.stories.tsx:40]: **FLAG** — [no-hardcoded-colors (self-referential)] The InsideCard story hardcodes hex colors in JSX style props: `border: '1px solid #e5e5e5'`, `color: '#666'`, `color: '#888'`. These are exactly the values the newly-shipped `no-hardcoded-colors` rule flags. Replace with Amplify CSS tokens (e.g. `var(--amplify-border-default)`, `var(--amplify-surface-secondary)`). Carried over from previous review — still unresolved.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Substantial design system expansion: new `@amplify/tokens-marketing` package, `@amplify/design-lint` ESLint plugin, semantic CSS token layers for tokens-foundation, and Storybook stories for 15+ UI components. Work is well-structured and the CI integration is solid. Two pre-existing issues from the previous round remain unresolved.**

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**2 FLAGs** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The `prefers-reduced-motion` override in both `tokens-foundation/src/motion.css` and `tokens-marketing/src/motion.css` collapses all durations to 1ms and staggers to 0ms — exactly the right accessibility treatment.
- Backward-compatible shadow token migration: old `sm`/`md`/`lg`/`sm-dark`/`md-dark`/`lg-dark` keys are preserved as legacy aliases alongside the new numeric elevation scale, so existing consumers won't break.
- The `design-lint` package correctly ships both a `recommended` (warn) and a `strict` (error) flat config preset — exactly the right ergonomic split for progressive adoption.
- CI wires `node --test packages/design-lint/test/*.test.js` and adds `tokens-marketing` to the build + dist verification loop in the same PR — the new packages won't silently drift from CI coverage.
- The `no-raw-spacing` rule correctly allows numeric `0` (semantically 'no spacing') and `var(--…)` / CSS-variable strings, avoiding the most common false positives.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/ui/src/components/MetricCard/MetricCard.stories.tsx (hunk 1 above)
Comment (zenith-oi) [inline packages/ui/src/components/MetricCard/MetricCard.stories.tsx:75]: **FLAG** — [no-raw-spacing (self-referential)] Still unresolved from the previous review: `gap: 16` in the Grid story's JSX style prop is a raw numeric pixel value — exactly what the newly-hardened `no-raw-spacing` rule flags. Since this PR ships the rule and the story in the same diff, the design system's own Storybook becomes a false-positive generator for any engineer who enables the strict preset. Replace with `gap: 'var(--amp-spacing-md)'` or an equivalent token.

File: packages/ui/src/components/Separator/Separator.stories.tsx (hunk 2 above)
Comment (zenith-oi) [inline packages/ui/src/components/Separator/Separator.stories.tsx:40]: **FLAG** — [no-hardcoded-colors (self-referential)] Still unresolved from the previous review: the InsideCard story hardcodes `#e5e5e5`, `#666`, and `#888` in JSX style props — exactly what the expanded `no-hardcoded-colors` Literal visitor now catches. The design system's own stories should use CSS variables. Replace with appropriate `--amplify-border-*` / `--amplify-text-*` tokens introduced in this same PR.

File: packages/tokens-marketing/src/gradients.css [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/**
+ * @amplify/tokens-marketing — Gradient palette
+ *
+ * Reusable gradient tokens — designed to hold up at hero scale and degrade
+ * gracefully on lower-quality screens. Use as `background-image:` values.
+ */
+:root {
+  /* Brand-aligned gradients */
+  --amplify-marketing-gradient-violet:
+    linear-gradient(135deg, #6531FF 0%, #A855F7 100%);
```
Comment (zenith-oi) [inline packages/tokens-marketing/src/gradients.css:10]: **FLAG** — [no-hardcoded-colors (self-referential)] The gradient definitions use raw hex literals (`#6531FF`, `#A855F7`, `#F59E0B`, `#FBBF24`, `#059669`, `#10B981`, `#E11D48`, `#F43F5E`) instead of referencing the `--amp-color-*` primitive variables already defined in `tokens-foundation`. While CSS custom properties can't be used inside `linear-gradient()` stop positions in all older browsers, modern browsers (and this system's stated targets) fully support `linear-gradient(135deg, var(--amp-color-violet-600) 0%, var(--amp-color-violet-400) 100%)`. Hardcoded hex values here mean the gradient tokens drift silently if the brand colour palette changes. Reference the primitives.

File: packages/eslint-config/package.json [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
+    "test": "node --test test/*.test.js"
+  },
   "peerDependencies": {
     "eslint": ">=8.0.0"
   },
+  "devDependencies": {
+    "eslint": "^9.0.0"
+  },
   "files": [
     "index.js",
```
Comment (zenith-oi) [inline packages/eslint-config/package.json:21]: **FLAG** — [missing-test-file-export] The `test/` directory containing `rules.test.js` is not listed in the `files` array (only `index.js`, `strict.js`, and `rules/` are included). Tests won't ship in the published package, which is usually correct, but the CI command `node --test packages/eslint-config/test/*.test.js` runs against the workspace source — that's fine. However if consumers try to run the tests from the installed package they can't. This is minor but worth confirming the intent: if tests are workspace-only (correct), add a comment; if they were meant to be published for extending teams, add `test/` to `files`.

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

**This PR adds a new `tokens-marketing` package, expands `tokens-foundation` with a Phase A CSS semantic layer (motion, shadows, surfaces, borders, fluid type), refactors and strengthens the three core ESLint rules, adds a `strict` preset, adds unit tests for all ESLint rules, and adds Storybook stories for 15 UI components. The work is solid overall.**

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**4 FLAGs** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The ESLint rule rewrites are meaningfully better: the old `no-hardcoded-colors` had two duplicate `JSXAttribute` handlers (only the second would ever run in the old visitor map) — the new `Literal` + `TemplateElement` approach is both correct and broader in coverage.
- The `strict.js` preset is a clean, additive pattern: it re-uses the same plugin instance exported from `index.js` rather than re-instantiating rules, so version drift between recommended and strict is impossible.
- `prefers-reduced-motion` overrides are present and correct in both `tokens-foundation` motion.css and `tokens-marketing` motion.css — including zeroing out parallax multipliers, which is thoughtful.
- Keeping `sm`/`md`/`lg` shadow aliases alongside the new numeric scale in `shadows.json` (with explicit DEPRECATED notes on old motion tokens) is the right backwards-compatible approach.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*
